### PR TITLE
CV2-5628: fix OR condition and remove projects filter

### PR DESCRIPTION
**Pull request by melsawy**
## Description

Remove projects filter and fix `OR` conditions .
- [x] Revisit ES `OR` conditions
- [x] Remove `projects` filter
- [x] Refactor build_feed_conditions method

References: CV2-5628

## How has this been tested?

Re-run  automated tests.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: lib/check_search.rb [hunk 1]
```diff
@@ -279,41 +272,50 @@ def get_search_field
...
+      custom_conditions.concat doc_conditions
+      custom_conditions.concat has_claim_conditions
+      custom_conditions.concat file_filter
+      custom_conditions.concat range_filter(:es)
+      custom_conditions.concat numeric_range_filter
+      custom_conditions.concat language_conditions
+      custom_conditions.concat fact_check_language_conditions
+      custom_conditions.concat request_language_conditions
+      custom_conditions.concat report_language_conditions
+      custom_conditions.concat team_tasks_conditions
```
[the remark this answers is not in the document]
Comment (melsawy) [reply lib/check_search.rb:301]: Yes @caiosba just move the block under `unless`

File: lib/check_search.rb [hunk 2]
```diff
@@ -409,25 +411,6 @@ def adjust_es_window_size
...
-
-      # Invalidate the search if empty... otherwise, adjust the projects filter
-      @options['projects'] = project_ids.empty? ? [0] : project_ids
-    end
-    if Team.current && !feed_query? && [@options['team_id']].flatten.size == 1
-      t = Team.find(team_id)
-      @options['projects'] = @options['projects'].blank? ? (Project.where(team_id: t.id).map(&:id) + [nil]) : Project.where(id: @options['projects'], team_id: t.id).map(&:id)
-    end
-    @options['projects'] += [nil] if @options['none_project']
-  end
```
[the remark this answers is not in the document]
Comment (melsawy) [reply lib/check_search.rb:429]: Good idea to make sure we always have a workspace filter

[the remark this answers is not in the document]
Comment (melsawy) [reply lib/check_search.rb:438]: @caiosba we already set a team filter and also return `[ProjectMedia.none` if the team filter is blank](url) so no need to add extra work for team filter

**Review by caiosba — CHANGES_REQUESTED** [review]
Sawy, please add the `adjust_team_work` to replace the `adjust_project_filter`, as we discussed.

**Comment by melsawy** [comment]
> Sawy, please add the `adjust_team_filter` method to replace the `adjust_project_filter` method you removed, as we discussed.

@caiosba already I found a [team_conditions methos](https://github.com/meedan/check-api/blob/develop/lib/check_search.rb#L72-L80) which do the same things and medias method return `ProjectMedia.none` if the the team filter is black so I think this is OK, what do you think?

**Comment by caiosba** [comment]
> > Sawy, please add the `adjust_team_filter` method to replace the `adjust_project_filter` method you removed, as we discussed.
> 
> @caiosba already I found a [team_conditions methos](https://github.com/meedan/check-api/blob/develop/lib/check_search.rb#L72-L80) which do the same things and medias method return `ProjectMedia.none` if the the team filter is black so I think this is OK, what do you think?

This is just returning the team values, I think it's important to make sure it's always part of the query.

**Comment by melsawy** [comment]
> > > Sawy, please add the `adjust_team_filter` method to replace the `adjust_project_filter` method you removed, as we discussed.
> > 
> > 
> > @caiosba already I found a [team_conditions methos](https://github.com/meedan/check-api/blob/develop/lib/check_search.rb#L72-L80) which do the same things and medias method return `ProjectMedia.none` if the the team filter is black so I think this is OK, what do you think?
> 
> This is just returning the team values, I think it's important to make sure it's always part of the query.

yes it's always part of the query either PG or ES

**Review by caiosba — APPROVED** [review]

**Pull request merged**